File: Integrated Sample Application/UPI.vb
Imports QRCoder

Public Class UPI
    Private Shared Function GetPaymentURL(ByVal vpa As String, ByVal payeeName As String, ByVal Optional txnId As String = "", ByVal Optional amount As Double? = Nothing, ByVal Optional currencyCode As String = "INR", ByVal Optional txnNote As String = "") As String
        If amount Is Nothing Then
            Return $"upi://pay?cu={currencyCode}&pa={vpa}&pn={payeeName}&tr={txnId}&am=0&mode=01&purpose=10&orgid=-&sign=-&tn={txnNote}"
        End If

        Return $"upi://pay?cu={currencyCode}&pa={vpa}&pn={payeeName}&tr={txnId}&am={amount}&mode=01&purpose=10&orgid=-&sign=-&tn={txnNote}"
    End Function

    Public Shared Function Generate(ByVal bank As Bank) As Bitmap
        Dim qrGenerator As QRCodeGenerator = New QRCodeGenerator()
        Dim qrCodeData As QRCodeData = qrGenerator.CreateQrCode(GetPaymentURL(vpa:=$"{bank.AccountNo}@{bank.IfscCode}.ifsc.npci", payeeName:=bank.PayeeName, amount:=bank.Amount, currencyCode:="INR", txnNote:=bank.TxnNote), QRCodeGenerator.ECCLevel.Q)
        Dim qrCode As QRCode = New QRCode(qrCodeData)
        Dim qrCodeImage As Bitmap = qrCode.GetGraphic(20)
        Return qrCodeImage
    End Function

    Public Shared Function Generate(ByVal vpa As VPA) As Bitmap
        Dim qrGenerator As QRCodeGenerator = New QRCodeGenerator()
        Dim qrCodeData As QRCodeData = qrGenerator.CreateQrCode(GetPaymentURL(vpa:=vpa.VpaId, payeeName:=vpa.PayeeName, amount:=vpa.Amount, currencyCode:="INR", txnNote:=vpa.TxnNote), QRCodeGenerator.ECCLevel.Q)
        Dim qrCode As QRCode = New QRCode(qrCodeData)
        Dim qrCodeImage As Bitmap = qrCode.GetGraphic(20)
        Return qrCodeImage
    End Function
End Class

File: DevNet.UPI/UPI.cs
using QRCoder;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DevNet
{
    public class UPI
    {

        private static string GetPaymentURL(string vpa, string payeeName, string txnId = "", double? amount = null, string currencyCode = "INR", string txnNote = "") 
        {
            if (amount == null)
            {
                return $"upi://pay?cu={currencyCode}&pa={vpa}&pn={payeeName}&tr={txnId}&am=0&mode=01&purpose=10&orgid=-&sign=-&tn={txnNote}";
            }
            return $"upi://pay?cu={currencyCode}&pa={vpa}&pn={payeeName}&tr={txnId}&am={amount}&mode=01&purpose=10&orgid=-&sign=-&tn={txnNote}";
        }
        public static Bitmap Generate(Models.Bank bank)
        {
            QRCodeGenerator qrGenerator = new QRCodeGenerator();
            QRCodeData qrCodeData = qrGenerator.CreateQrCode(GetPaymentURL(
                vpa: $"{bank.AccountNo}@{bank.IfscCode}.ifsc.npci",
                payeeName: bank.PayeeName,
                amount: bank.Amount,
                currencyCode: "INR",
                txnNote: bank.TxnNote
                ), QRCodeGenerator.ECCLevel.Q);
            QRCode qrCode = new QRCode(qrCodeData);
            Bitmap qrCodeImage = qrCode.GetGraphic(20);
            return qrCodeImage;
        }

        public static Bitmap Generate(Models.VPA vpa)
        {
            QRCodeGenerator qrGenerator = new QRCodeGenerator();
            QRCodeData qrCodeData = qrGenerator.CreateQrCode(GetPaymentURL(
                vpa: vpa.VpaId,
                payeeName: vpa.PayeeName,
                amount: vpa.Amount,
                currencyCode: "INR",
                txnNote: vpa.TxnNote
                ), QRCodeGenerator.ECCLevel.Q);
            QRCode qrCode = new QRCode(qrCodeData);
            Bitmap qrCodeImage = qrCode.GetGraphic(20);
            return qrCodeImage;
        }
    }
}
